syntax = "proto3";

package src;

import "src/brinson.proto";
import "src/am_sma.proto";
import "src/backtest.proto";


option go_package=".;src";


service MicroService {
  rpc BrinsonHandler(brinson.Request) returns (brinson.Response);

  rpc SMAPortfolio(sma.NULL) returns (stream sma.Portfolio);

  rpc SMAPortfolioExpanded(sma.NULL) returns (stream sma.PortfolioExpanded);

  rpc SMABalance(sma.Request) returns (stream sma.Balance);

  rpc SMABalanceExpanded(sma.Request) returns (stream sma.BalanceExpanded);

  rpc SMAIncome(sma.Request) returns (stream sma.Income);

  rpc SMAIncomeAsset(sma.Request) returns (stream sma.IncomeAsset);

  rpc SMAHolding(sma.Request) returns (stream sma.Holding);

  rpc SMATransaction(sma.Request) returns (stream sma.Transaction);

  rpc SMADetailFee(sma.Request) returns (stream sma.DetailFee);

  rpc SMASecurity(sma.Request) returns (stream sma.Security);

  rpc SMASecurityQuote(sma.Request) returns (stream sma.SecurityQuote);

  rpc SMABenchmark(sma.Request) returns (stream sma.Benchmark);

  rpc SMAInterestTax(sma.Request) returns (stream sma.InterestTax);

  rpc Backtest(backtest.Request) returns (backtest.Response);
}
